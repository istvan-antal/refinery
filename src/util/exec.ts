/* eslint-disable no-new-func */
const exec = (params: { [key: string]: {} }, code: string) => {
    try {
        return (new Function(
            ...Object.keys(params),
            `return ${code}`,
        ))(
            ...Object.values(params),
        );
    } catch (error) {
        return { error: error.toString() };
    }
};

export default exec;