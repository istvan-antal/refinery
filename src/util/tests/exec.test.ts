import exec from '../exec';

test('exec', () => {
    expect(exec({ data: [] }, 'data')).toEqual([]);
});

test('exec with error', () => {
    expect(exec({ data: [] }, '[')).toMatchSnapshot();
});