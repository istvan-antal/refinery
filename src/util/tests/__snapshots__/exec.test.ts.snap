// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exec with error 1`] = `
Object {
  "error": "SyntaxError: Unexpected token '}'",
}
`;
